Src/Plugins/Tests.Security/EndToEndTests.cs

```diff
@@ -28,22 +28,28 @@ protected EndToEndTests(ITestOutputHelper outputHelper) : base(outputHelper)
 
         protected abstract string RuleId { get; }
 
+        protected abstract string Service { get; }
+
         protected abstract string Framework { get; }
 
         protected override string TestLogResourceNameRoot => $"Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security.TestData.{TypeUnderTest}";
 
-        protected override string ProductDirectory => Path.Combine(base.ProductDirectory, @"Plugins\Tests.Security");
+        protected override string TestBinaryTestDataDirectory => Path.Combine(ProductRootDirectory, "src", "Plugins", TestBinaryName, "TestData");
+
+        protected override string ProductTestDataDirectory => Path.Combine(TestBinaryTestDataDirectory, Service);
 
         protected override IDictionary<string, string> ConstructTestOutputsFromInputResources(IEnumerable<string> inputResourceNames, object parameter)
         {
             var results = new Dictionary<string, string>();
             var dict = new Dictionary<string, Task<string>>();
 
+            const string input = "Inputs.";
+
             foreach (string inputResourceName in inputResourceNames)
             {
-                string secret = inputResourceName.Substring("Inputs.".Length);
+                string name = inputResourceName.Substring(inputResourceName.IndexOf(input) + input.Length);
```

Comment:
>inputResourceName [](http://example.com/codeflow?start=58&length=17)

Previously, the inputResourceName was composed of `Inputs.RuleId.FileName.extension`. With the sarif-sdk change, inputResourceName has the fully qualified name: `Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security. etc etc`.

So, I had to made a change to find the name of the file that we are analyzing.

Reply:
I took a deeper look at this.
The Sarif.Sdk changed the behavior in the following way:
- previously, we had:
```csharp
        private string ConstructFullInputResourceName(string resourceName)
            => "Inputs." + resourceName;
```

- currently, we have:
```csharp
        private string GetFullResourcePathForInputResource(string resourceName)
        {
            return $"{TestLogResourceNameRoot}.Inputs.{resourceName}";
        }
```

There isn't any API that converts the embedded resource path to the file name. Also, we don't see this issue in the sarif.sdk because almost all the tests uses the RunTest call passing a single file, which wouldn't generate this.

Reply:
To reduce this hard-coded strings, I'm now using the parameter argument and pass the original list. It is not ideal, but that will prevent us from hard-coding any substring/indexof logic.